zone/spell_effects.cpp

```diff
@@ -1820,6 +1820,58 @@ bool Mob::SpellEffect(Mob* caster, uint16 spell_id, float partial, int level_ove
 
 				break;
 			}
+			case SE_SummonCorpseZone:
+			{
+				if (IsClient()) {
+					Client* client_target = this->CastToClient();
+					if(client_target->IsGrouped()) {
+						Group* group = client_target->GetGroup();
+						if(!group->IsGroupMember(caster)) {
+							if (caster != this) {
+								caster->MessageString(Chat::Red, SUMMON_ONLY_GROUP_CORPSE);	
+								break;
+							}
+						}
+					} else if (caster) {
+						if(caster->IsRaidGrouped()) {
+							Raid *raid = caster->GetRaid();
+							uint32 group_id = raid->GetGroup(caster->GetName());
+							if(group_id > 0 && group_id < MAX_RAID_GROUPS) {
+								if(raid->GetGroup(client_target->GetName()) != group_id) {
+									caster->MessageString(Chat::Red, SUMMON_ONLY_GROUP_CORPSE);
+									break;
+								}
+							}
+						} else {
+							if(caster != this) {
+								caster->MessageString(Chat::Red, SUMMON_ONLY_GROUP_CORPSE);
+								break;
+							}
+						}
+					}
+					
+					if(client_target) {
```

Comment:
Styling: any reason we have scrunched if blocks around here, but not in other places?

Reply:
Simply because I have an if/else and not just one statement. Would have included the { } elsewhere as well if it were more than a single conditional.

Reply:
I'm just talking about `if(client_target) ` versus `if (client_target) `

Reply:
Ah, okay. No particular reason. Just missed the space I guess. :P